.github/workflows/hermetic_library_generation.yaml

```diff
@@ -0,0 +1,54 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# GitHub action job to test core java library features on
+# downstream client libraries before they are released.
+name: Hermetic library generation upon generation config change through pull requests
+on:
+  pull_request:
+
+jobs:
+  library_generation:
+    # skip pull requests come from a forked repository
+    if: github.event.pull_request.head.repo.full_name == github.repository
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+        token: ${{ secrets.CLOUD_JAVA_BOT_TOKEN }}
+    - name: Build image
+      shell: bash
+      run: |
+        docker build \
```

Comment:
Compare to the workflow in google-cloud-java, is this the only main change?

Reply:
Changes compared to google-cloud-java
1. build local image
2. install gapic-generator-java-pom-parent

Reply:
An idea to get rid of the `.m2` volume mapping.
Our workflows build and publish a docker image every time we merge a PR in this repo. The tag is the commit SHA of the merged (squashed) commit. 

Would it be useful to run `mvn install -pl ggj-pom-parent` in the Dockerfile and label the image as `x.x.x-SNAPSHOT` as well (we can have two tags)?

Reply:
This step is specific to generate common protos. 

Putting `mvn install` in the Dockerfile will apply to all libraries, which is a redundant step.